src/NLog/Layouts/XmlAttribute.cs

```diff
@@ -0,0 +1,111 @@
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.Layouts
+{
+    using NLog.Config;
+    using NLog.Internal;
+
+    /// <summary>
+    /// JSON attribute.
+    /// </summary>
+    [NLogConfigurationItem]
+    [ThreadAgnostic]
+    [ThreadSafe]
+    public class XmlAttribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmlAttribute" /> class.
+        /// </summary>
+        public XmlAttribute() : this(null, null, true) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmlAttribute" /> class.
+        /// </summary>
+        /// <param name="name">The name of the attribute.</param>
+        /// <param name="layout">The layout of the attribute's value.</param>
+        public XmlAttribute(string name, Layout layout) : this(name, layout, true) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmlAttribute" /> class.
+        /// </summary>
+        /// <param name="name">The name of the attribute.</param>
+        /// <param name="layout">The layout of the attribute's value.</param>
+        /// <param name="encode">Encode value with json-encode</param>
+        public XmlAttribute(string name, Layout layout, bool encode)
+        {
+            Name = name;
+            Layout = layout;
+            Encode = encode;
+            IncludeEmptyValue = false;
+            LayoutWrapper.XmlEncodeNewlines = true;
+        }
+
+        /// <summary>
+        /// Gets or sets the name of the attribute.
+        /// </summary>
+        /// <docgen category='XML Attribute Options' order='10' />
+        [RequiredParameter]
+        public string Name { get => _name; set => _name = XmlHelper.XmlConvertToElementName(value?.Trim(), true); }
+        private string _name;
+
+        /// <summary>
+        /// Gets or sets the layout that will be rendered as the attribute's value.
+        /// </summary>
+        /// <docgen category='XML Attribute Options' order='10' />
+        [RequiredParameter]
+        public Layout Layout
+        {
+            get => LayoutWrapper.Inner;
+            set => LayoutWrapper.Inner = value;
+        }
+
+        /// <summary>
+        /// Determines wether or not this attribute will be Json encoded.
+        /// </summary>
+        /// <docgen category='XML Attribute Options' order='100' />
+        public bool Encode
```

Comment:
maybe default it to true, as 2 of the 3 ctors set it to true

Reply:
The default is true, as it gets the default value from `XmlEncodeLayoutRendererWrapper`